src/main/java/views/main_menu/MainMenuV.java

```diff
@@ -1,35 +1,158 @@
+        this.saveButton.setAlignmentX(Component.CENTER_ALIGNMENT);
+        this.saveButton.setAlignmentY(Component.CENTER_ALIGNMENT);
+        this.createMonthButton.setAlignmentX(Component.CENTER_ALIGNMENT);
+        this.createMonthButton.setAlignmentY(Component.CENTER_ALIGNMENT);
+
+        this.add(new JPanel());
         this.add(title);
+        this.add(new JPanel());
+        this.add(monthSelection);
+        this.add(new JPanel());
+        this.add(this.createMonthButton);
+        this.add(new JPanel());
+        this.add(this.saveButton);
+        this.add(new JPanel());
+
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
         this.setVisible(false);
     }
 
     /**
-     * Makes the main menu visible which displays the session data passed and displays a message in a popup
+     * Makes the main menu visible which displays the session data, and opens a popup with the message
+     *
      * @param message a message to display in a popup after opening the menu
-     * @param session the session data to display
+     * @param session the session data to display, or null if displaying a message is not needed
      */
     @Override
     public void openMainMenu(String message, SessionStorage session) {
+        this.session = session;
+
+        ArrayList<MonthlyStorage> monthlyData = session.getAllMonthlyData();
+        ArrayList<Integer> monthIDs = new ArrayList<>();
+
+        for (MonthlyStorage month : monthlyData) {
+            monthIDs.add(month.getMonthID());
+        }
+
+        // Add month ids to dropdown menu to display
+        this.selectMonthComboBox.removeAllItems();
+        for (Integer id : monthIDs) {
+            this.selectMonthComboBox.addItem(id.toString());
+        }
+
+        // Only enable dropdown and select buttons if we do have months to display
+        this.selectMonthComboBox.setEnabled(!monthIDs.isEmpty());
+        this.selectMonthButton.setEnabled(!monthIDs.isEmpty());
+
         this.setVisible(true);
-        JOptionPane.showMessageDialog(this, message);
-        System.out.println(session);
+        if (message != null) {
+            JOptionPane.showMessageDialog(this, message);
+        }
+    }
+
+    /**
+     * Reacts to UI events
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource() == this.selectMonthComboBox) {
+            String selectedID = (String) this.selectMonthComboBox.getSelectedItem();
+            // This can't be null because dropdown is disabled when there are no months available
+            assert selectedID != null;
+            this.selectedMonthID = Integer.parseInt(selectedID);
+        } else if (e.getSource() == this.selectMonthButton) {
+            UpdateViewC monthMenuController = new UpdateViewC(new UpdateViewUCI(new MonthMenuP()));
+            new MonthMenuV(monthMenuController, this.session, this.selectedMonthID);
```

Comment:
Just a reminder that you will need to call MonthMenuV's openMonthMenu() after my branch gets merged.

Reply:
Yep 👍